base/float.jl

```diff
@@ -78,11 +78,35 @@ for T in (Float16, Float32, Float64)
     @eval exponent_bits(::Type{$T}) = $(sizeof(T)*8 - significand_bits(T) - 1)
     @eval exponent_bias(::Type{$T}) = $(Int(exponent_one(T) >> significand_bits(T)))
     # maximum float exponent
-    @eval exponent_max(::Type{$T}) = $(Int(exponent_mask(T) >> significand_bits(T)) - exponent_bias(T))
+    @eval exponent_max(::Type{$T}) = $(Int(exponent_mask(T) >> significand_bits(T)) - exponent_bias(T) - 1)
     # maximum float exponent without bias
     @eval exponent_raw_max(::Type{$T}) = $(Int(exponent_mask(T) >> significand_bits(T)))
 end
 
+"""
+    exponent_max(T)
```

Comment:
@fredrikekre  I hesitate to add these docstring (rebased and modified from the previous PR) it was suggested to add for clarification. My concern: we don't export this method, so does documenting the method raise an issue? I know we prefix the module name for public API  that are  *not* exported but still part of the formal API. In this instance, I don't recall what the policy is, but it does seem helpful to include some documentation even if they are primarily internal methods. Should I convert them to comments instead, perhaps?

Reply:
We document internal functions all over the place. More documentation is always better.

Reply:
Yes, this was deliberate